sdk/servicebus/azure-servicebus/azure/servicebus/_pyamqp/aio/_link_async.py

```diff
@@ -226,6 +232,9 @@ async def _outgoing_detach(self, close: bool = False, error: Optional[AMQPError]
         await self._session._outgoing_detach(detach_frame) # pylint: disable=protected-access
         if close:
             self._is_closed = True
+            self.links.pop(self.name, None)
```

Comment:
Had to add this here - see Test `test_queue_message_settle_through_mgmt_link_due_to_broken_receiver_link` was causing settlement over mgmt to not happen bc we weren't recognizing that the receiver link had been closed